crates/bioauth-flow-rpc/src/lib.rs

```diff
@@ -309,49 +406,34 @@ where
     async fn enroll(&self, liveness_data: LivenessData) -> RpcResult<()> {
         self.deny_unsafe.check_if_safe()?;
 
-        info!("Bioauth flow - enrolling in progress");
+        self.do_enroll(liveness_data).await.map_err(EnrollError)?;
 
-        let public_key = rpc_validator_key_logic::validator_public_key(&self.validator_key_extractor).map_err(EnrollError::KeyExtraction)?;
-        let (opaque_liveness_data, signature) = self.sign(public_key.clone(), &liveness_data).await
-            .map_err(EnrollError::Sign)?;
+        Ok(())
+    }
 
-        self.robonode_client
-            .as_ref()
-            .enroll(EnrollRequest {
-                liveness_data: opaque_liveness_data.as_ref(),
-                liveness_data_signature: signature.as_ref(),
-                public_key: public_key.as_ref(),
-            })
+    async fn enroll_v2(&self, liveness_data: LivenessData) -> RpcResult<EnrollV2Result> {
+        self.deny_unsafe.check_if_safe()?;
+
+        let EnrollResponse { scan_result_blob } = self
+            .do_enroll(liveness_data)
             .await
-            .map_err(EnrollError::Robonode)?;
+            .map_err(EnrollV2Error)?;
 
-        info!("Bioauth flow - enrolling complete");
+        let scan_result_blob = scan_result_blob.expect("expect scan result blob in response");
```

Comment:
Avoid panics or error after the action has been successfully conducted

Reply:
https://github.com/humanode-network/humanode/pull/1026/commits/9738337651606636a4761d190a4bc93aee37967d